src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
+        temp_f2 = (temp_a3->scale.x * 50.0f) + 6.0f;
+
+        if (!(temp_f2 < fabsf(sp20.x))) {
+            temp_f0 = fabsf(sp20.z);
+            if (!(temp_f2 < temp_f0)) {
+                return 0;
+            }
+        }
+    } else {
+    	return 0;
+    }
+
+    return 1;
+}
+
+
+
+void ObjWarp2block_SwapWithChild(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Vec3f tempVec;
+    Vec3s tempRot;
+    s32 temp;
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.world.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.world.pos, &this->dyna.actor.child->world.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.child->world.pos, &tempVec);
+
+    temp = this->dyna.actor.world.rot.y;
+    this->dyna.actor.world.rot.y = this->dyna.actor.child->world.rot.y;
+    this->dyna.actor.child->world.rot.y = temp;
+
+    temp = this->dyna.actor.shape.rot.y;
+    this->dyna.actor.shape.rot.y = this->dyna.actor.child->shape.rot.y;
+    this->dyna.actor.child->shape.rot.y = temp;
+
+    temp = this->dyna.actor.home.rot.z;
+    this->dyna.actor.home.rot.z = this->dyna.actor.child->home.rot.z;
+    this->dyna.actor.child->home.rot.z = temp;
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.scale);
+    Math_Vec3f_Copy(&this->dyna.actor.scale, &this->dyna.actor.child->scale);
+    Math_Vec3f_Copy(&this->dyna.actor.child->scale, &tempVec);
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.focus.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.focus.pos, &this->dyna.actor.child->focus.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.child->focus.pos, &tempVec);
+
+    tempRot = this->dyna.actor.focus.rot;
+    this->dyna.actor.focus.rot = this->dyna.actor.child->focus.rot;
+    this->dyna.actor.child->focus.rot = tempRot;
+
+    temp = this->dyna.actor.params & 0x7FFF;
+    this->dyna.actor.params = (this->dyna.actor.params & 0x8000) | (this->dyna.actor.child->params & 0x7FFF);
+    this->dyna.actor.child->params = (this->dyna.actor.child->params & 0x8000) | (temp  & 0x7FFF);
+
+    if (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &this->dyna.actor.home.pos) < 0.01f) {
+        Flags_UnsetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+    } else {
+    	Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+    }
+}
+
+s32 func_80BA2218(ObjWarp2block *this, GlobalContext *globalCtx) {
+	Player* player = PLAYER;
+
+    if (func_80BA1ECC(this, globalCtx)) {
+
+        if (player->stateFlags2 & 0x1000000) {
+            func_8010BD58(globalCtx, 1);
+            this->actionFunc_168 = func_80BA228C;
+        } else {
+            player->stateFlags2 |= 0x800000;
+        }
+    }
+
+    return 0;
+}
+
+
+s32 func_80BA228C(ObjWarp2block *this, GlobalContext *globalCtx) {
+    if (globalCtx->msgCtx.unk_E3EE == 4) {
+        this->actionFunc_168 = func_80BA2218;
+    }
+
+    if (globalCtx->msgCtx.unk_E3EC == 0xA) {
+        if (this->unk_172 == 0xFE) {
+            this->unk_16E = 0x6E;
+        } else {
+	        this->unk_16E = this->unk_16E - 1;
```

Comment:
```suggestion
	        this->unk_16E--;
```

Reply:
It's also possible to move this operation to a pre-decrement directly in the conditional `if (--this->unk_16E == )`. Whatever you think is more readable.

Reply:
we dont have a set standard for that, but most seem to prefer a separated decrement vs an inline predecrement. and i would have to agree

Reply:
Sounds good!